### ISSUE TITLE
Documentation Request + Private Field Discussion

### ISSUE THREAD

**Issue by Zamiell** [mbilski/exhaustivestruct#7]
Related to: #1

Hello and thanks for the excellent linter!

1) Can you document the fact that it only works on public fields in the README.md? It was very frustrating that I just had to figure this out through trial and error. Wasted about 15 minutes of my time.

2) Your fix to #1 was to just make it ignore **all** private fields, but I believe a more nuanced approach is required. Since the point of the linter is to ensure that fields are not skipped during initialization, structs with private fields are now bug-prone.

Consider the following code:

```go
// This is version 1 of my Foo program
package foo1

type Foo struct {
	Field1 int
	Field2 int
	Field3 int
}

func New(field1Value int) *Foo {
	return &Foo{
		Field1: field1Value,
		Field2: 200,
		// Crap, I forgot to initialize Field3, and now my program is bugged.
		// Thank heavens that the exhaustivestruct linter found this bug for me!
		// Mateusz Bilski is a god!
	}
}
```

```go
// This is version 2 of my Foo program
package foo2

type Foo struct {
	Field1 int
	// I realized that field2 and field3 didn't need to be public,
	// so I refactored them to be private for better safety!
	field2 int
	field3 int
}

func New(field1Value int) *Foo {
	return &Foo{
		Field1: field1Value,
		field2: 200,
		// Crap, I forgot to initialize field3, and now my program is bugged.
		// exhaustivestruct does not warn me about this code, even though I would expect it to.
		// Press the button here: http://nooooooooooooooo.com/
	}
}
```

Thus, I propose that exhaustivestruct should ignore private fields if they are being referenced from OUTSIDE of the package, like the example code in #1 shows. But if they are being referenced from INSIDE the package, then they should be included in the checks.

**Comment by mbilski** [comment]
fixed in #10

**Comment by Zamiell** [comment]
Excellent, thank you Mateusz!
